.clang-tidy

```diff
@@ -13,6 +13,10 @@ bugprone-*,
 -bugprone-reserved-identifier,
 -bugprone-swapped-arguments,
 -bugprone-unchecked-optional-access,
+clang-analyzer-core.*,
+clang-analyzer-cplusplus.*,
```

Comment:
Why should double check this doesn't cause a massive speed regression on CI due to any expensive new checks.